### PR TITLE
Clean up build guide and docs after net6 removal

### PR DESCRIPTION
**Pull request by mdaigle**
Restructures the buildguide document. Adds a reference section and a workflows section. Previously, the doc was structured as a collection of example commands, but it didn't clearly call out dependencies (e.g. building tests before running them and how to do so for that type of test run) and was a poor way to discover and understand the MSBuild targets and parameters.

Also includes small doc and test updates to remove .NET6 references.

### CODE REVIEW

**Review by cheenamalhotra — APPROVED** [review]

File: src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/DataStreamTest/DataStreamTest.cs [hunk 1]
```diff
@@ -1556,14 +1556,10 @@ private static void ReadStream(string connectionString)
                             DataTestUtility.AssertThrowsWrapper<ArgumentOutOfRangeException>(() => stream.Read(buffer, -1, 2));
                             DataTestUtility.AssertThrowsWrapper<ArgumentOutOfRangeException>(() => stream.Read(buffer, 2, -1));
 
-                            // Prior to net6 comment:ArgumentException is thrown in net5 and earlier. ArgumentOutOfRangeException in net6 and later
```
Comment (cheenamalhotra) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/DataStreamTest/DataStreamTest.cs:1559]: Prior to .NET 6 includes .NET Framework as well right?

Comment (mdaigle) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/DataStreamTest/DataStreamTest.cs:1559]: I think you're right, I reverted this.

**Review by samsharma2700 — APPROVED** [review]

**Pull request merged**